js/video-player-embed.js

```diff
@@ -1,99 +1,199 @@
+      if (!streamState) {
+        live.play();
+        streamState = 1;
+      }
+    }
+
+    if (options.uri.indexOf('/ipfs/') > 0) {
+      document.getElementById('loadingTitle').innerHTML = 'Located stream via IPFS';
+      document.getElementById('msg').innerHTML = 'Downloading video content...';
+      // Use specified IPFS gateway by replacing it in the uri
+      options.uri = ipfs_gateway + options.uri.substring(options.uri.indexOf('/ipfs/'));
+
+      // Wait for two .ts chunks to be loaded before applying seek action
+      if (streamState < 3) {
+        streamState++;
+        if (streamState == 3) {
+          if (!start_from) {
+            // Seek to live after waiting 1 s
+            setTimeout(function() { live.liveTracker.seekToLiveEdge(); }, 1);
+          } else {
+            // Seek to start_from time after waiting 1 s
+            setTimeout(function() { live.currentTime(start_from); }, 1);
+          }
+        }
+      }
+    }
+
+    if (options.uri.indexOf('/ipns/') > 0) {
+      document.getElementById('loadingTitle').innerHTML = 'Located stream via IPFS';
+      document.getElementById('msg').innerHTML = 'Downloading video content...';
+      options.uri = ipfs_gateway + options.uri.substring(options.uri.indexOf('/ipns/'));
+    }
+    console.debug(options.uri);
+    return options;
+  };
+}
+
+function loadStream() {
+  document.getElementById('loadingStream').style.display = 'block';
+  document.getElementById('selectStream').style.display = 'none';
+}
+
+document.querySelector('.ipfs-stream').addEventListener('click', function(event) {
+  ipfsStream();
+});
+
+document.querySelector('.http-stream').addEventListener('click', function(event) {
+  httpStream();
+});
+
+live.metadata = 'none';
+
+live.on('loadedmetadata', function() {
+  document.getElementById('streamSelector').style.display = 'none';
+});
+
+live.on('loadeddata', function(event) {
+  console.debug(event);
+});
+
+var refreshButton = document.createElement('button');
+refreshButton.className = 'button button-primary compact stream-refresh';
+refreshButton.innerHTML = 'Refresh page and try again';
+refreshButton.addEventListener('click', function() {
+  window.location.reload(true);
+});
+
+live.on('error', function(event) {
+  console.debug(this.error());
+  document.getElementById('loadingTitle').innerHTML = 'Unable to load video stream';
+  document.querySelector('.loader-animation').style.display = 'none';
+  document.getElementById('msg').innerHTML = this.error().message;
+  document.getElementById('loadingStream').appendChild(refreshButton);
+});
+
+if (!stream_urls_http || !Array.isArray(stream_urls_http) || (stream_urls_http.length === 0)) {
+  document.querySelector('.http-stream').setAttribute('disabled', 'disabled');
+}
+
+// Video sharing links
+var rootURL = window.location.href.split('?')[0];
+
+function getShareLink(key) {
+  var bookmark = live.currentTime(); // TODO Change to use ipfs hash

```

Comment:
@darkdrgn2k this needs to be the ipfs hash

Reply:
https://github.com/tomeshnet/ipfs-live-streaming/compare/carriage...ipfsHash?expand=1
`getCurrentHash(live.currentTime())` should do it